### PR TITLE
Add support to the user profile page to specify an email

### PR DESCRIPTION
**Pull request by maxime-poulain**
This commit introduces the possibility for users to specify their email.
 * A new guard clause is added to validate emails.
 * A new rule builder extension on string to validate email has been added.
 * Imported [EmailValidator](https://github.com/jstedfast/EmailValidation/blob/master/EmailValidation/EmailValidator.cs) class, this make us independent of the NuGet.
 * The library passes the following  [tests](https://github.com/jstedfast/EmailValidation/blob/master/UnitTests/EmailValidationTests.cs)

### CODE REVIEW

File: src/Smart.FA.Catalog.Application/Extensions/FluentValidation/OnStringRuleBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+using FluentValidation;
+using Smart.FA.Catalog.Core.Helpers;
+
+namespace Smart.FA.Catalog.Application.Extensions.FluentValidation;
+
+public static class OnStringRuleBuilderExtensions
+{
+    /// <summary>
+    /// Defines a rules to validate a email format.
+    /// THe rule checks if the email is null or empty, then validates the format of the email.
```
Comment (BomUbik) [inline src/Smart.FA.Catalog.Application/Extensions/FluentValidation/OnStringRuleBuilderExtensions.cs:10]: `THe` -> `The`

File: src/Smart.FA.Catalog.Application/Extensions/FluentValidation/OnStringRuleBuilderExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+using FluentValidation;
+using Smart.FA.Catalog.Core.Helpers;
+
+namespace Smart.FA.Catalog.Application.Extensions.FluentValidation;
+
+public static class OnStringRuleBuilderExtensions
+{
+    /// <summary>
+    /// Defines a rules to validate a email format.
```
Comment (BomUbik) [inline src/Smart.FA.Catalog.Application/Extensions/FluentValidation/OnStringRuleBuilderExtensions.cs:9]: `a email` -> `an email`

File: src/Smart.FA.Catalog.Core/SeedWork/Guard.cs [hunk 3]
```diff
@@ -22,4 +23,15 @@ public static void AgainstNull(object? argumentValue, string argumentName)
         if (argumentValue is null)
             throw new ArgumentNullException(argumentName);
     }
+
+    public static string AgainstInvalidEmail(string? email, string parameter, string? message = null)
```
Comment (BomUbik) [inline src/Smart.FA.Catalog.Core/SeedWork/Guard.cs:27]: A parameter named `parameter` is a bit confusing. What about `parameterName` or `emailParameterName` ?

**Review by BomUbik — APPROVED** [review]

File: src/Smart.FA.Catalog.Core/SeedWork/Guard.cs (hunk 3 above)
Comment (maxime-poulain) [reply src/Smart.FA.Catalog.Core/SeedWork/Guard.cs:27]: Usually you name that parameterName indeed, as you can see here:
https://github.com/ardalis/GuardClauses/blob/main/src/GuardClauses/GuardAgainstNullExtensions.cs

I'll rename.

**Pull request merged**